R/occurrenceChange.r

```diff
@@ -70,19 +72,40 @@
 #' }                   
 #' @export
 
-occurrenceChange <- function(firstYear, lastYear, bayesOut, change = 'growthrate'){
+occurrenceChange <- function(firstYear, lastYear, bayesOut, change = 'growthrate', region = NULL){
   
+  # error checks for years
   if(!firstYear %in% bayesOut$min_year:bayesOut$max_year) stop('firstYear must be in the year range of the data')
   if(!lastYear %in% bayesOut$min_year:bayesOut$max_year) stop('lastYear must be in the year range of the data')
   
+  # error checks for change
   if(!class(change) == 'character') stop('Change must be a character string identifying the change metric.  Either: difference, percentdif, growthrate or lineargrowth')
-  
   if(!change %in% c('difference', 'percentdif', 'growthrate', 'lineargrowth')) stop('The change metric must be one of the following: difference, percentdif, growthrate or lineargrowth')
   
-  occ_it <- bayesOut$BUGSoutput$sims.list$psi.fs
+  # error check for region
+  if(!class(region) == 'character') stop('region must be a character string identifying the regional estimates that change is to be calculated for.')
+  
+  
+  
+  
+  # extract the sims list, if there is a region code, use the psi.fs for that region
+  if(!is.null(region)){
+  reg_code <- paste("psi.fs.r_", region, sep = "")
+  
+  occ_it <- bayesOut$BUGSoutput$sims.list
+  occ_it <- occ_it[[grep(reg_code, names(occ_it))]]
```

Comment:
what happens if they spell the region wrong? you might need an error check for that here